Sprint_6 UI tests with POM and allure results

Yulia7052 opened the pull request.

Yulia7052 committed:
Sprint_6 UI tests with POM and allure results

AlexKitIt commented on pages/base_page.py:
```diff
@@ -0,0 +1,43 @@
+        self.find_element(button).click()
+
+    def get_text(self, element):
+        WebDriverWait(self.driver, 3).until(expected_conditions.visibility_of_element_located((element[0], element[1])))
```
Метод используется несколько раз - значит есть смысл сделать его отдельным

AlexKitIt commented on pages/main_page_helper.py:
```diff
@@ -0,0 +1,63 @@
+    def __init__(self, driver):
+        super().__init__(driver)
+
+    def click_header_order_button(self):
```
Нужно исправить: здесь и далее: над каждым методом пейджи ставится аннотация allure.step

AlexKitIt commented on pages/main_page_helper.py:
```diff
@@ -0,0 +1,63 @@
+
+    def click_middle_order_button(self):
+        element = self.driver.find_element(*self.middle_order_button)
+        self.driver.execute_script('arguments[0].scrollIntoView(true);', element)
```
Нужно исправить: здесь и далее по проекту: все прямые обращения к driver необходимо вынести в base и оттуда за счет наследования переиспользовать в объектах страниц

AlexKitIt commented on tests/test_check_make_order.py:
```diff
@@ -0,0 +1,64 @@
+
+class TestOrder:
+
+    def test_check_make_order_from_header(self, driver):
```
Нужно исправить: здесь и далее: над каждым тестовым методом ставится аннотация allure.title

AlexKitIt commented on tests/test_check_make_order.py:
```diff
@@ -0,0 +1,64 @@
+
+        main_helper.click_header_order_button()
+
+        order_helper.set_name("Вупсень")
```
Можно лучше: тестовые данные стоит хранить отдельно в data модуле, так поддерживать и переиспользовать будет проще

AlexKitIt commented on tests/test_check_tab_logo_scooter.py:
```diff
@@ -0,0 +1,22 @@
+        helper.get_main_page()
+        helper.click_header_scooter_logo()
+
+        assert driver.current_url == "https://qa-scooter.praktikum-services.ru/"
```
Нужно исправить: здесь и далее: прямого обращения к driver из тестов быть не должно, только через методы пейджей

AlexKitIt commented on tests/test_check_tab_questions.py:
```diff
@@ -0,0 +1,35 @@
+    @pytest.mark.parametrize(
+        'question_number,question_text',
+        [
+            [1, 'Сутки — 400 рублей. Оплата курьеру — наличными или картой.'],
+            [2,
+             'Пока что у нас так: один заказ — один самокат. Если хотите покататься с друзьями, можете просто сделать '
+             'несколько заказов — один за другим.'],
+            [3,
+             'Допустим, вы оформляете заказ на 8 мая. Мы привозим самокат 8 мая в течение дня. Отсчёт времени аренды '
+             'начинается с момента, когда вы оплатите заказ курьеру. Если мы привезли самокат 8 мая в 20:30, '
+             'суточная аренда закончится 9 мая в 20:30.'],
+            [4, 'Только начиная с завтрашнего дня. Но скоро станем расторопнее.'],
+            [5, 'Пока что нет! Но если что-то срочное — всегда можно позвонить в поддержку по красивому номеру 1010.'],
+            [6,
+             'Самокат приезжает к вам с полной зарядкой. Этого хватает на восемь суток — даже если будете кататься '
+             'без передышек и во сне. Зарядка не понадобится.'],
+            [7, 'Да, пока самокат не привезли. Штрафа не будет, объяснительной записки тоже не попросим. Все же свои.'],
+            [8, 'Да, обязательно. Всем самокатов! И Москве, и Московской области.']
```
Можно лучше: тексты ответов стоит хранить снаружи в data модулях для поддержания и актуализации

Yulia7052 committed:
Sprint_6 UI tests with POM

modify annotation allure.title and allure.step; modify usage driver in tests and classes

AlexKitIt commented on pages/main_page_helper.py:
```diff
@@ -0,0 +1,71 @@
+    @allure.step('Кликаем на кнопку "Заказать" в средней части страницы')
+    def click_middle_order_button(self):
+        element = self.driver.find_element(*self.middle_order_button)
+        self.driver.execute_script('arguments[0].scrollIntoView(true);', element)
```
Повторный комент
